src/components/HdBadge.vue

```diff
@@ -0,0 +1,217 @@
+      },
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    showIconBefore: {
+      type: Boolean,
+      default: true,
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+  },
+  data() {
+    return {
+      collapsed: false,
+    };
+  },
+  computed: {
+    computedClasses() {
+      const baseClass = 'badge';
+      const classes = [baseClass];
+
+      if (this.modifier) {
+        classes.push(`${baseClass}--${this.modifier}`);
+      }
+      if (this.hasDetails) {
+        classes.push(`${baseClass}--with-details`);
+      }
+      if (this.collapsed) {
+        classes.push(`${baseClass}--collapsed`);
+      }
+
+      return classes;
+    },
+    hasDetails() {
+      return !!this.$slots.default;
+    },
+  },
+  methods: {
+    collapse() {
+      if (this.hasDetails) this.collapsed = !this.collapsed;
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.badge{
+    display: flex;
+    justify-content: flex-start;
+
+    &__label{
+      display: flex;
+      align-items: center;
+      font-weight: bold;
+      font-size: 18px;
+      line-height: 28px;
+      border-radius: 4px;
+      padding: $stack-xs $inline-s;
+      cursor: default;
+
+      &::after{
+        content: ' ';
+        display: none;
+        background: url('~hd-blocks/assets/icons/ic_chevron-down--small.svg');
+        width: 12px;
+        height: 7px;
+        margin-left: $inline-m;
+        margin-right: $inline-s;
+      }
+    }
+
+    ::v-deep  &__icon{
+      width: 14px;
+      height: 14px;
+      margin-right: $inline-s;
+    }
+
+    &__details{
+      visibility: hidden;
+      width: 170px;
+      white-space: nowrap;
+      overflow: hidden;
+
+      padding: $inset-s;
+      padding-top: 0;
+      border-bottom-left-radius: $inline-xs;
+      border-bottom-right-radius: $inline-xs;
+      border-top-left-radius: 0;
+      border-top-right-radius: 0;
+      font-weight: normal;
+      font-size: 18px;
+      line-height: 28px;
+    }
+
+    &--primary{
+        color: $mantis;
+
+        .badge__label, .badge__details{
+          background: rgba($mantis, 0.3);
+        }
+    }
+
+    &--secondary{
+        color: $vivid-blue;
+
+        .badge__label, .badge__details{
+          background: rgba($vivid-blue, 0.3);
+        }
+    }
+
+    &--tertiary{
+        color: $crusta;
+
+        .badge__label, .badge__details{
+          background: rgba($crusta, 0.3);
+        }
+    }
+
+    &--disabled{
+        color: $nevada;
+
+        .badge__label, .badge__details{
+          background: rgba($heather, 0.3);
+        }
+    }
+
+    &--with-details > &__label{
+      cursor: pointer;
+
+      &::after{
+        display: block;
+      }
+    }
+
+    &--collapsed{
+      display: block;
+    }
+
+    &--collapsed > &__label{
+      border-bottom-left-radius: 0;
+      border-bottom-right-radius: 0;
+
+      &::after{
+        background: url('~hd-blocks/assets/icons/ic_chevron-up.svg');
```

Comment:
And here we can just rotate it
```suggestion
        transform: rotateZ(180deg);
```